WordPress/WordPressApi/WordPressComRestApi.swift

```diff
@@ -0,0 +1,251 @@
+import Foundation
+import AFNetworking
+
+/**
+ Error constants for the WordPress.com REST API
+
+ - InvalidInput:                   The parameters sent to the server where invalid
+ - InvalidToken:                   The token provided was invalid
+ - AuthorizationRequired:          Permission required to access resource
+ - UploadFailed:                   The upload failed
+ - Unknown:                        Unknow error happen
+ */
+public enum WordPressComRestApiError: Int, ErrorType {
+    case InvalidInput
+    case InvalidToken
+    case AuthorizationRequired
+    case UploadFailed
+    case Unknown
+}
+
+public final class WordPressComRestApi: NSObject
+{
+    public typealias SuccessResponseBlock = (responseObject: AnyObject, httpResponse: NSHTTPURLResponse?) -> ()
+    public typealias FailureReponseBlock = (error: NSError, httpResponse: NSHTTPURLResponse?) -> ()
+
+    private static let apiBaseURLString: String = "https://public-api.wordpress.com/rest/"
+
+    private let oAuthToken: String
+
+    private lazy var sessionManager: AFHTTPSessionManager = {
+        let baseURL = NSURL(string:WordPressComRestApi.apiBaseURLString)
+        let sessionConfiguration = NSURLSessionConfiguration.defaultSessionConfiguration()
+        sessionConfiguration.HTTPAdditionalHeaders = ["Authorization": "Bearer \(self.oAuthToken)"]
+        let sessionManager = AFHTTPSessionManager(baseURL:baseURL, sessionConfiguration:sessionConfiguration)
+        sessionManager.responseSerializer = WordPressComRestAPIResponseSerializer()
+        sessionManager.requestSerializer = AFJSONRequestSerializer()
+        return sessionManager
+    }()
+
+    public init(oAuthToken: String) {
+        self.oAuthToken = oAuthToken
+    }
+
+    deinit {
+        sessionManager.invalidateSessionCancelingTasks(true);
+    }
+    /**
+     Reset the API instance
+
+     Invalidates the session and cancel all pending requests
+     */
+    public func reset() {        
+        sessionManager.invalidateSessionCancelingTasks(true);
+    }
+
+    //MARK: - Network requests
+
+    /**
+     Executes a GET request to the specified endpoint defined on URLString
+
+     - parameter URLString:  the url string to be added to the baseURL
+     - parameter parameters: the parameters to be encoded on the request
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the upload and to cancel the upload
+     */
+    public func GET(URLString: String,
+                     parameters:[NSString:AnyObject]?,
+                     success:SuccessResponseBlock,
+                     failure:FailureReponseBlock) -> NSProgress?
```

Comment:
I'm willing to 👍 this for this PR, as it's probably out of scope, but I found the optional return `NSProgress` confusing. I understand that AFNetworking might return a `nil` task if the request can't be serialized. At the very least, the documentation should explain in which conditions it can return `nil`, and that in that case the request won't be sent and neither of the completion blocks will be called.

An even better approach would be to return a non-optional NSProgress instead, and `throw` the serialization if it happens. I think that makes it way more obvious that it's an error case, and will allow callers to handle it better.